### ISSUE TITLE
Currently failing unit tests in the GitHub Actions pipeline

### ISSUE THREAD

**Issue by Lucas-C** [py-pdf/fpdf2#1021]
I have just found that there are 46 unit tests that are failing in our test suite:
```
test/encryption/test_encryption.py::test_encrypt_font FAILED             [ 35%]
test/encryption/test_encryption.py::test_encryption_unicode FAILED       [ 35%]
test/fonts/test_add_font.py::test_add_font_with_str_fname_ok FAILED      [ 36%]
test/fonts/test_add_font.py::test_render_en_dash FAILED                  [ 37%]
test/fonts/test_add_font.py::test_add_font_otf FAILED                    [ 37%]
test/fonts/test_charmap.py::test_charmap_first_999_chars[Quicksand-Regular.otf] FAILED [ 37%]
test/fonts/test_charmap.py::test_charmap_first_999_chars[DroidSansFallback.ttf] FAILED [ 37%]
test/fonts/test_charmap.py::test_charmap_first_999_chars[DejaVuSans-Oblique.ttf] FAILED [ 37%]
test/fonts/test_charmap.py::test_charmap_first_999_chars[Garuda.ttf] FAILED [ 37%]
test/fonts/test_charmap.py::test_charmap_first_999_chars[cmss12.ttf] FAILED [ 37%]
test/fonts/test_charmap.py::test_charmap_first_999_chars[DejaVuSansMono.ttf] FAILED [ 37%]
test/fonts/test_charmap.py::test_charmap_first_999_chars[DejaVuSans.ttf] FAILED [ 38%]
test/fonts/test_charmap.py::test_charmap_first_999_chars[Roboto-Regular.ttf] FAILED [ 38%]
test/fonts/test_font_fallback.py::test_fallback_font FAILED              [ 38%]
test/fonts/test_font_fallback.py::test_fallback_font_ignore_style FAILED [ 38%]
test/fonts/test_font_remap.py::test_emoji_glyph FAILED                   [ 38%]
test/fonts/test_font_remap.py::test_nb_replace FAILED                    [ 38%]
test/fonts/test_font_remap.py::test_two_mappings FAILED                  [ 39%]
test/fonts/test_font_remap.py::test_thai_text FAILED                     [ 39%]
test/html/test_html.py::test_issue_156 FAILED                            [ 40%]
test/html/test_html.py::test_html_heading_hebrew FAILED                  [ 40%]
test/html/test_html.py::test_html_custom_pre_code_font FAILED            [ 41%]
test/outline/test_outline.py::test_toc_with_russian_heading FAILED       [ 53%]
test/table/test_table.py::test_table_with_ttf_font FAILED                [ 81%]
test/table/test_table.py::test_table_with_ttf_font_and_headings FAILED   [ 81%]
test/text/test_cell.py::test_cell_markdown_with_ttf_fonts FAILED         [ 88%]
test/text/test_cell.py::test_cell_markdown_right_aligned FAILED          [ 88%]
test/text/test_cell.py::test_cell_curfont_leak FAILED                    [ 89%]
test/text/test_multi_cell.py::test_multi_cell_justified_with_unicode_font FAILED [ 92%]
test/text/test_multi_cell.py::test_multi_cell_j_paragraphs FAILED        [ 93%]
test/text/test_multi_cell.py::test_multi_cell_font_leakage FAILED        [ 93%]
test/text/test_multi_cell.py::test_multi_cell_font_stretching FAILED     [ 93%]
test/text/test_multi_cell.py::test_multi_cell_char_spacing FAILED        [ 93%]
test/text/test_multi_cell_markdown.py::test_multi_cell_markdown_with_ttf_fonts FAILED [ 94%]
test/text/test_text.py::test_text_positioning FAILED                     [ 95%]
test/text/test_varied_fragments.py::test_varfrags_fonts FAILED           [ 96%]
test/text/test_write.py::test_write_font_stretching FAILED               [ 97%]
test/text_shaping/test_text_shaping.py::test_indi_text FAILED            [ 98%]
test/text_shaping/test_text_shaping.py::test_mixed_text_shaping FAILED   [ 98%]
test/text_shaping/test_text_shaping.py::test_text_replacement FAILED     [ 98%]
test/text_shaping/test_text_shaping.py::test_kerning PASSED              [ 98%]
test/text_shaping/test_text_shaping.py::test_hebrew_diacritics FAILED    [ 98%]
test/text_shaping/test_text_shaping.py::test_ligatures FAILED            [ 98%]
test/text_shaping/test_text_shaping.py::test_multi_cell_markdown_with_shaping FAILED [ 98%]
test/text_shaping/test_text_shaping.py::test_features FAILED             [ 99%]
test/text_shaping/test_text_shaping.py::test_text_with_parentheses FAILED [ 99%]
```

Example GitHub Action job log: https://github.com/py-pdf/fpdf2/actions/runs/6867529148/job/18676573426

We should try to figure out **why, and what caused this problem**.
So far, I have just found that all the failing tests have in common a failing MD5 hash comparison.

The last successfull pipeline execution on the `master` branch was yesterday:
https://github.com/py-pdf/fpdf2/actions/runs/6858623125

**Comment by Lucas-C** [comment]
I found the culprit: there has been a new `fonttools` release 5 hours ago:
https://pypi.org/project/fonttools/#history

**Comment by Lucas-C** [comment]
Sadly, this `fonttools==4.44.1` seems to include some backward incompatible changes:
```
_____________________________________________ test_charmap_first_999_chars[cmss12.ttf] ______________________________________________
test/conftest.py:99: in assert_pdf_equal
    actual_pdf.output(expected.open("wb"), linearize=linearize)
fpdf/fpdf.py:4930: in output
    self.buffer = output_producer.bufferize()
fpdf/output.py:378: in bufferize
    font_objs_per_index = self._add_fonts()
fpdf/output.py:575: in _add_fonts
    subsetter.subset(font.ttfont)
/home/lucas_cimon/.local/share/virtualenvs/fpdf2/lib/python3.8/site-packages/fontTools/subset/__init__.py:3498: in subset
    self._prune_post_subset(font)
/home/lucas_cimon/.local/share/virtualenvs/fpdf2/lib/python3.8/site-packages/fontTools/subset/__init__.py:3459: in _prune_post_subset
    old_codepages = font[tag].getCodePageRanges()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <'OS/2' table at 7ff795a5fbb0>

    def getCodePageRanges(self):
        """Return the set of 'ulCodePageRange*' bits currently enabled."""
        bits = set()
>       ul1, ul2 = self.ulCodePageRange1, self.ulCodePageRange2
E       AttributeError: 'table_O_S_2f_2' object has no attribute 'ulCodePageRange1'
...
====================================================== short test summary info ======================================================
FAILED test/fonts/test_charmap.py::test_charmap_first_999_chars[cmss12.ttf] - AttributeError: 'table_O_S_2f_2' object has no attribute 'ulCodePageRange1'
FAILED test/html/test_html.py::test_html_HTMLMixin_deprecation_warning - assert 2 == 1
FAILED test/table/test_table.py::test_table_with_ttf_font - AttributeError: 'table_O_S_2f_2' object has no attribute 'ulCodePageRange1'
```

**Comment by Lucas-C** [comment]
I opened PR https://github.com/py-pdf/fpdf2/pull/1022 as a work-in-progress solution

**Comment by Lucas-C** [comment]
This has been fixed by #1025
